metabase/metabase
# setup loki visual testing for echarts

Closes https://github.com/metabase/metabase/issues/37803

### Description

Adds Loki for visual testing. It consists of two workflows GHA:
- `loki.yml`: runs loki tests on CI and reports about errors
- `loki-update.yml`: when `loki-update` label is added to a PR the workflow updates snapshots and pushes the changes

Why `loki-update` label instead of a comment? For simplicity now since all changes to the comment-based workflow will have to be merged in `master` to work.

You can run loki tests locally using `yarn test-visual:loki`. It will put failed diff tests to `./loki/` folder.

### How to verify

Check this PR: https://github.com/metabase/metabase/pull/37777

- Link to failed run with artifacts: https://github.com/metabase/metabase/pull/37777#issuecomment-1896123943
- Updated snapshots after adding `loki-update`label https://github.com/metabase/metabase/pull/37777#issuecomment-1896159454
